### PR TITLE
Created Login component with Redux integration

### LINKED ISSUE #11: Create login component

### PR DESCRIPTION
**Pull request by bsuro10**

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
+import Login from './components/Login';
 
 import Routing from './Routing';
 
 function App() {
   return (
-    <Routing/>
+    // <Routing/>
```
Comment (PolaDinitz) [inline client/src/App.tsx]: remove pls

File: client/src/components/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        setValue(e.target.value);
+    }
+
+    return {
+        value,
+        onChange: handleChange
+    }
+}
+
+export default Login;
```
Comment (PolaDinitz) [inline client/src/components/Login.tsx]: no newline

File: client/src/config/config.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const config = {
+    apiUrl: "http://localhost:3000"
+};
```
Comment (PolaDinitz) [inline client/src/config/config.tsx]: no newline

File: client/src/index.tsx [hunk 4]
```diff
@@ -2,11 +2,17 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
+import {Provider} from 'react-redux';
```
Comment (PolaDinitz) [inline client/src/index.tsx]: { provide } - spaces

File: client/src/redux/action-types/user.action-types.tsx [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const userActionTypes = {
+    LOGIN_REQUEST: 'USER_LOGIN_REQUEST',
+    LOGIN_SUCCESS: 'USER_LOGIN_SUCCESS',
+    LOGIN_FAILED: 'USER_LOGIN_FAILED',
+    LOGOUT: 'USER_LOGOUT',
+}
```
Comment (PolaDinitz) [inline client/src/redux/action-types/user.action-types.tsx]: no newline

File: client/src/redux/reducers/user.reducer.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { userActionTypes } from '../action-types/user.action-types';
+
+let user = JSON.parse(localStorage.getItem('user') || '{}');
+const initialState = user !== '{}' ? { loggedIn: true, user } : {loggedIn: true};
```
Comment (PolaDinitz) [inline client/src/redux/reducers/user.reducer.tsx]: make an object to contain more than just a user

File: client/src/components/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                </Typography>
+            </Box>
+        </Paper>
+    );
+}
+
+const useFormInput = (initialValue: String) => {
+    const [value, setValue] = useState(initialValue);
+
+    const handleChange = (e: any) => {
```
Comment (PolaDinitz) [inline client/src/components/Login.tsx]: type: Event<>

File: client/src/components/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                    fullWidth
+                    required
+                    sx={{margin: "10px"}}
+                />
+                <Button 
+                    disabled={loading} 
+                    sx={{margin: "10px"}} 
+                    type="submit" 
+                    variant="contained"
+                    onClick={handleLogin}>{loading ? 'Loging...' : 'Login'} </Button>
```
Comment (PolaDinitz) [inline client/src/components/Login.tsx]: format.
```js
 <Button 
                    disabled={loading} 
                    sx={{margin: "10px"}} 
                    type="submit" 
                    variant="contained"
                    onClick={handleLogin}>
                               {loading ? 'Loging...' : 'Login'}
</Button>
```

File: client/src/components/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
+import { Avatar, TextField, Paper, InputAdornment, Button, Typography, Link } from '@mui/material';
+import EmailIcon from '@mui/icons-material/Email';
+import LoginIcon from '@mui/icons-material/Login';
```
Comment (PolaDinitz) [inline client/src/components/Login.tsx]: combine imports

File: client/src/App.tsx [hunk 10]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
+import Login from './components/Login';
 
 import Routing from './Routing';
```
Comment (PolaDinitz) [inline client/src/App.tsx]: redundant import

**Review by PolaDinitz — CHANGES_REQUESTED** [review]

File: client/src/components/Login.tsx (hunk 7 above)
Comment (bsuro10) [reply client/src/components/Login.tsx]: `ChangeEvent<HTMLInputElement>` worked, is that fine?

Comment (PolaDinitz) [reply client/src/components/Login.tsx]: yes

**Pull request merged**